### ISSUE TITLE
landsat notebook failing with ValueError: Unsupported input type: res_(x: <class 'tuple'>)

### ISSUE THREAD

**Issue by rsignell-usgs** [Element84/geo-notebooks#8]
Tried running the landsat notebook today and failed in cell [9]:
```python
dc = stac_load(item_collection,
               bands=['red', 'blue', 'green', 'nir08'],
               chunks={"x": 2048, "y": 2048},
               output_crs=output_crs,
               resolution=resolution,
               groupby='solar_day',
               stac_cfg=cfg
)
```
with
```python-traceback
/home/conda/users/d4dd112dfad7cb570b71d7c58e7679057fd3d160eeb5781ffdfb4fdad9d96def-20220622-155312-091009-65-pangeo/lib/python3.9/site-packages/odc/stac/_mdtools.py:549: UserWarning: Missing asset with name: atmos_opacity
  warn(f"Missing asset with name: {band}")
/home/conda/users/d4dd112dfad7cb570b71d7c58e7679057fd3d160eeb5781ffdfb4fdad9d96def-20220622-155312-091009-65-pangeo/lib/python3.9/site-packages/odc/stac/_mdtools.py:549: UserWarning: Missing asset with name: cloud_qa
  warn(f"Missing asset with name: {band}")

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File <timed exec>:14, in <module>

File /home/conda/users/d4dd112dfad7cb570b71d7c58e7679057fd3d160eeb5781ffdfb4fdad9d96def-20220622-155312-091009-65-pangeo/lib/python3.9/site-packages/odc/stac/_load.py:460, in load(items, bands, groupby, resampling, dtype, chunks, pool, crs, resolution, align, geobox, bbox, lon, lat, x, y, like, geopolygon, progress, stac_cfg, patch_url, preserve_original_order, **kw)
    457 items = list(items)
    458 _parsed = list(parse_items(items, cfg=stac_cfg))
--> 460 gbox = output_geobox(
    461     _parsed,
    462     bands=bands,
    463     crs=crs,
    464     resolution=resolution,
    465     align=align,
    466     geobox=geobox,
    467     like=like,
    468     geopolygon=geopolygon,
    469     bbox=bbox,
    470     lon=lon,
    471     lat=lat,
    472     x=x,
    473     y=y,
    474 )
    476 if gbox is None:
    477     raise ValueError("Failed to auto-guess CRS/resolution.")

File /home/conda/users/d4dd112dfad7cb570b71d7c58e7679057fd3d160eeb5781ffdfb4fdad9d96def-20220622-155312-091009-65-pangeo/lib/python3.9/site-packages/odc/stac/_mdtools.py:797, in output_geobox(items, bands, crs, resolution, align, geobox, like, geopolygon, bbox, lon, lat, x, y)
    795 x0, y0, x1, y1 = _compute_bbox(items, crs)
    796 geopolygon = geom.box(x0, y0, x1, y1, crs)
--> 797 return GeoBox.from_geopolygon(geopolygon, resolution=resolution, align=align)

File /home/conda/users/d4dd112dfad7cb570b71d7c58e7679057fd3d160eeb5781ffdfb4fdad9d96def-20220622-155312-091009-65-pangeo/lib/python3.9/site-packages/odc/geo/geobox.py:226, in GeoBox.from_geopolygon(geopolygon, resolution, crs, align, shape, tight, anchor, tol)
    223 else:
    224     geopolygon = geopolygon.to_crs(crs)
--> 226 return GeoBox.from_bbox(
    227     geopolygon.boundingbox,
    228     crs,
    229     shape=shape,
    230     resolution=resolution,
    231     anchor=anchor,
    232     tol=tol,
    233     tight=tight,
    234 )

File /home/conda/users/d4dd112dfad7cb570b71d7c58e7679057fd3d160eeb5781ffdfb4fdad9d96def-20220622-155312-091009-65-pangeo/lib/python3.9/site-packages/odc/geo/geobox.py:145, in GeoBox.from_bbox(bbox, crs, tight, shape, resolution, anchor, tol)
    142     shape = None
    144 if resolution is not None:
--> 145     rx, ry = res_(resolution).xy
    146     if _snap is None:
    147         offx, nx = snap_grid(bbox.left, bbox.right, rx, None, tol=tol)

File /home/conda/users/d4dd112dfad7cb570b71d7c58e7679057fd3d160eeb5781ffdfb4fdad9d96def-20220622-155312-091009-65-pangeo/lib/python3.9/site-packages/odc/geo/types.py:291, in res_(x)
    289 if isinstance(x, (int, float)):
    290     return Resolution(float(x))
--> 291 raise ValueError(f"Unsupported input type: res_(x: {type(x)})")

ValueError: Unsupported input type: res_(x: <class 'tuple'>)
```

**Comment by gadomski** [comment]
I can reproduce. I'll see what I can do about a fix.

**Comment by gadomski** [comment]
The core problem is upstream in `odc-stac`, I've opened a PR to fix the issue: https://github.com/opendatacube/odc-stac/pull/77. However, a workaround should be possible to get this notebook working in the interim. I'll look into that now.

**Comment by Kirill888** [comment]
workaround is documented here: 
https://github.com/opendatacube/odc-stac/issues/74#issuecomment-1166175426

**Comment by gadomski** [comment]
@rsignell-usgs in addition to the fix in opendatacube/odc-stac#77, there's two notebook fixes required to get things working. I've created a PR with those fixes here: https://github.com/Element84/geo-notebooks/pull/9 -- your review would be appreciated.
